library/model/user.py

```diff
@@ -32,20 +38,22 @@ def __init__(self, email, firstname, lastname, mob1, mob2, area_code, landline,
         self.read_books = []
         self.invoices = []
 
-    def borrow_book(self, book: Book) -> Optional[BorrowedBook]:
-        try:
-            if book.can_borrow():
-                borrowed_book = book.borrow_book()
-                self.borrowed_books.append(borrowed_book)
-                LibraryRepository.update_user(self)
-                return borrowed_book
-            return None
-        except AttributeError:
-            return None
-        except ValueError:
-            return None
+    def borrow_book(self, book: Book) -> Borrowing:
```

Comment:
You have extracted a new class borrowing, why not also extract the borrowing funcitonality to the respective class?